### PR TITLE
Ikuguk-added new test

### PR DESCRIPTION
**Pull request by kuguk**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/rozetka/components/CatalogMenu.java [hunk 1]
```diff
@@ -25,4 +26,21 @@ public CategoryPage openCategory(String categoryName) {
                 .shouldHave(sizeGreaterThanOrEqual(1))
                 .texts();
     }
+
+    @Step("CatalogMenu: open products list {subCategoryName} through hovered category {hoveredCategoryName}")
+    public ProductsListPage openProductsListByCategoryInCatalogMenu(String hoveredCategoryName, String... subCategoryName) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/components/CatalogMenu.java]: avoid varargs

Comment (toleksyn) [reply src/main/java/com/softserveinc/ita/rozetka/components/CatalogMenu.java]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java [hunk 2]
```diff
@@ -71,4 +73,43 @@ public void testPossibilityToFilterProductsByPriceRange() {
...
+        productsListPage.setCheckBoxFilter(firstFilterType, firstFilterValue);
+
+        var secondFilterType = "Колір корпусу";
+        var secondFilterValue = "Нержавіюча сталь";
+        productsListPage.setCheckBoxFilter(secondFilterType, secondFilterValue);
+
+        var productsCount = productsListPage.getProductsAmount();
+        assertNotEquals(productsCount, 0, "Incorrect products filters parameters");
+
+        SoftAssert softassert = new SoftAssert();
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java]: var

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java [hunk 3]
```diff
@@ -71,4 +73,43 @@ public void testPossibilityToFilterProductsByPriceRange() {
...
+        //next cycle runs 1, 2 times for 1-2 products, and 3 times for products count > 3 (checks first, middle and last product in filtered list)
+        for (var productNumber = 1; productNumber <= productsCount; ) {//productNumber = productNumber + productCount / 2) {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java]: ?

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java [hunk 4]
```diff
@@ -71,4 +73,43 @@ public void testPossibilityToFilterProductsByPriceRange() {
...
+            var characteristicTab = productsListPage
+                    .openProductByNumber(productNumber)
+                    .openProductTab(ProductPageTab.CHARACTERISTICS);
+            softassert.assertEquals(characteristicTab.getCharacteristicText(firstFilterType), firstFilterValue, "Incorrect product characteristic");
+            softassert.assertEquals(characteristicTab.getCharacteristicText(secondFilterType), secondFilterValue, "Incorrect product characteristic");
+
+            if (productNumber == 1 + productsCount / 2)
+                productNumber = productsCount;
+            else
+                productNumber = productNumber + productsCount / 2;
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java:115]: java code convention

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java (hunk 3 above)
Comment (kuguk) [reply src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java]: fixed

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java (hunk 4 above)
Comment (kuguk) [reply src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java:115]: fixed

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java (hunk 2 above)
Comment (kuguk) [reply src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java]: fixed

File: src/main/java/com/softserveinc/ita/rozetka/components/CatalogMenu.java (hunk 1 above)
Comment (kuguk) [reply src/main/java/com/softserveinc/ita/rozetka/components/CatalogMenu.java]: fixed

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java [hunk 5]
```diff
@@ -76,4 +77,45 @@ public void testPossibilityToFilterProductsByPriceRange() {
...
+            softAssert.assertEquals(characteristicTab.getCharacteristicText(secondFilterType), secondFilterValue, "Incorrect product characteristic");
+
+            if (productNumber == 1 + productsCount / 2) {
+                productNumber = productsCount;
+            } else {
+                productNumber = productNumber + productsCount / 2;
+            }
+            if (productsCount == 1) {
+                break;
+            }
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java:119]: this logic is not obvious

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java (hunk 5 above)
Comment (kuguk) [reply src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java:119]: fixed (added comment before cycle)

**Pull request merged**